staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go

```diff
@@ -1056,23 +927,20 @@ func (g *Cloud) ensureInternalForwardingRule(existingFwdRule *compute.Forwarding
 	}
 	// At this point, the existing rule has been deleted if required.
 	// Create the rule based on the api version determined
-	if newFRC.Version() == meta.VersionBeta {
-		klog.V(2).Infof("ensureInternalLoadBalancer(%v): creating beta forwarding rule", newFRC.name)
-		var betaRule *computebeta.ForwardingRule
-		betaRule, err = newFRC.ToBeta()
-		if err != nil {
-			return err
-		}
-		err = g.CreateBetaRegionForwardingRule(betaRule, g.region)
-	} else {
-		var gaRule *compute.ForwardingRule
-		klog.V(2).Infof("ensureInternalLoadBalancer(%v): creating ga forwarding rule", newFRC.name)
-		gaRule, err = newFRC.ToGA()
-		if err != nil {
-			return err
-		}
-		err = g.CreateRegionForwardingRule(gaRule, g.region)
+	klog.V(2).Infof("ensureInternalLoadBalancer(%v): creating forwarding rule", newFwdRule.Name)
+	if err = g.CreateRegionForwardingRule(newFwdRule, g.region); err != nil {
+		return err
 	}
-	klog.V(2).Infof("ensureInternalLoadBalancer(%v): created forwarding rule, err : %s", newFRC.name, err)
-	return err
+	klog.V(2).Infof("ensureInternalLoadBalancer(%v): created forwarding rule", newFwdRule.Name)
+	return nil
+}
+
+func forwardingRulesEqual(old, new *compute.ForwardingRule) bool {
+	return (old.IPAddress == "" || new.IPAddress == "" || old.IPAddress == new.IPAddress) &&
```

Comment:
Trying to understand the logic - isn't `old.IPAddress == new.IPAddress` sufficient?

Reply:
This logic was introduced in #46663 (Search for `FwdRuleEqual` in `gce_loadbalancer_internal.go`)

@prameshj brought this up as well, we are still figuring out the thought process behind this logic. Probably, I will fix this in a follow up PR.

Reply:
It seems fine to replace with just "old.IPAddress == new.IPAddress". I am thinking the `new.IPAddress == ""` is to disallow change when IP assignment drops to empty if network/subnetwork have not changed.

I can't see why `old.IPAddress== ""` needs to be here. We were thinking of fixing in a follow-up PR so that in can be easily reverted without reverting the GA api change.